src/lib/modes.js

```diff
@@ -112,3 +112,8 @@ export const METHOD_GUARD = {
   begin: '\\.\\s*' + UNDERSCORE_IDENT_RE,
   relevance: 0
 };
+
+export const END_FIRST_MATCH_SAME_AS_BEGIN = {
+  'after:begin': (m, resp) => { resp.data.heredoc = m[1]; },
+  'before:end': (m, resp) => { if (resp.data.heredoc !== m[1]) resp.ignoreMatch(); }
```

Comment:
Why `heredoc`?

Reply:
I guess now that I pulled it into a common module I should make the naming more specific.